src/password.cr

```diff
@@ -25,12 +25,21 @@ module Scrypt
       @parts = @raw_hash.split("$")
     end
 
-    def ==(password)
+    def verify(password : String)
       key_len = digest.bytesize / 2
       hashed_password = Engine.hash_secret(password, salt, key_len: key_len, cost: cost)
       Crypto::Subtle.constant_time_compare(@raw_hash.to_slice, hashed_password.to_slice)
     end
 
+    @[Deprecated("Use Scrypt::Password#verify instead")]
+    def ==(password : String)
+      verify password
+    end
+
+    def ==(other : Password)
+      digest == other.digest
+    end
```

Comment:
Just one last thing: can we drop this new equality method? Comparing the digest alone is wrong (what about cost or salt?), and I don't really see a use case.

Reply:
Sorry I didn't see this until now. I'm working on [this](https://github.com/dfabric/dppm-rest-api) project right now, which has users structured like [this](https://github.com/DFabric/dppm-rest-api/blob/e8fa4adf9d36384e5af65eb2d1a7abada33801be/spec/permissions.json#L113-L129). I would like to enable admins to select a user by the API key's hash -- but perhaps `password.to_s == input` is a better way to do that. The other reason is for testing purposes, but you can always just reopen it in that case. Thoughts?

Reply:
You can't rely on the password digest to be stable: the password can be reencrypted at any time, leading to a different digest.

The only stable equality was comparing the hashed password with the actual password string, but that's being removed for a formal verification, and you can't use the actual password for anything but verifying the password.

Reply:
hm, ok.